diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 2742e19..6a637ad 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -16,7 +16,6 @@ if (WIN32)
 	add_subdirectory(3rdparty/wil EXCLUDE_FROM_ALL)
 	add_subdirectory(3rdparty/D3D12MemAlloc EXCLUDE_FROM_ALL)
 else()
-	find_package(LibLZMA REQUIRED)
 	make_imported_target_if_missing(LibLZMA::LibLZMA LIBLZMA)
 
 	# Using find_package OpenGL without either setting your opengl preference to GLVND or LEGACY
