diff -uNr a/gfx/widgets/gfx_widget_achievement_popup.c b/gfx/widgets/gfx_widget_achievement_popup.c
--- a/gfx/widgets/gfx_widget_achievement_popup.c	2026-08-25 22:25:20.276467674 +0000
+++ b/gfx/widgets/gfx_widget_achievement_popup.c	2026-08-25 22:25:38.292581397 +0000
@@ -337,13 +337,13 @@
 
       /* Title */
       gfx_widgets_draw_text(
-         &p_dispwidget->gfx_widget_fonts.regular,
+         &p_dispwidget->gfx_widget_fonts.msg_queue,
          state->queue[state->queue_read_index].title,
          screen_pos_x + state->height - text_unfold_offset
          + p_dispwidget->simple_widget_padding,
          screen_pos_y
-         + p_dispwidget->gfx_widget_fonts.regular.line_height
-         + p_dispwidget->gfx_widget_fonts.regular.line_ascender,
+         + p_dispwidget->gfx_widget_fonts.msg_queue.line_height
+         + p_dispwidget->gfx_widget_fonts.msg_queue.line_ascender,
          video_width,
          video_height,
          TEXT_COLOR_FAINT,
@@ -354,13 +354,13 @@
 
       /* TODO: is a ticker necessary ? */
       gfx_widgets_draw_text(
-         &p_dispwidget->gfx_widget_fonts.regular,
+         &p_dispwidget->gfx_widget_fonts.msg_queue,
          state->queue[state->queue_read_index].subtitle,
          screen_pos_x + state->height - text_unfold_offset
          + p_dispwidget->simple_widget_padding,
          screen_pos_y + state->height
-         - p_dispwidget->gfx_widget_fonts.regular.line_height
-         - p_dispwidget->gfx_widget_fonts.regular.line_descender,
+         - p_dispwidget->gfx_widget_fonts.msg_queue.line_height
+         - p_dispwidget->gfx_widget_fonts.msg_queue.line_descender,
          video_width,
          video_height,
          TEXT_COLOR_INFO,
@@ -370,7 +370,7 @@
       if (is_folding)
       {
          gfx_widgets_flush_text(video_width, video_height,
-            &p_dispwidget->gfx_widget_fonts.regular);
+            &p_dispwidget->gfx_widget_fonts.msg_queue);
 
          if (dispctx && dispctx->scissor_end)
             dispctx->scissor_end(video_info->userdata,
@@ -568,13 +568,13 @@
    const dispgfx_widget_t* p_dispwidget = (const dispgfx_widget_t*)
       state->dispwidget_ptr;
 
-   state->height = p_dispwidget->gfx_widget_fonts.regular.line_height * 4.0f;
+   state->height = p_dispwidget->gfx_widget_fonts.msg_queue.line_height * 4.0f;
    state->width = MAX(
       font_driver_get_message_width(
-         p_dispwidget->gfx_widget_fonts.regular.font,
+         p_dispwidget->gfx_widget_fonts.msg_queue.font,
          state->queue[state->queue_read_index].title, strlen(state->queue[state->queue_read_index].title), 1.0f),
       font_driver_get_message_width(
-         p_dispwidget->gfx_widget_fonts.regular.font,
+         p_dispwidget->gfx_widget_fonts.msg_queue.font,
          state->queue[state->queue_read_index].subtitle, strlen(state->queue[state->queue_read_index].subtitle), 1.0f)
    );
    state->width += p_dispwidget->simple_widget_padding * 2;
