diff --git a/lib/crau/crau.h b/lib/crau/crau.h
index 0d4f9f1..b99283f 100644
--- a/lib/crau/crau.h
+++ b/lib/crau/crau.h
@@ -62,6 +62,10 @@
 # define CRAU_CONTEXT_STACK_DEPTH 3
 #endif /* CRAU_CONTEXT_STACK_DEPTH */
 
+#ifndef __has_c_attribute
+#define __has_c_attribute(x) 0
+#endif
+
 struct crau_context_stack_st {
 	long stack[CRAU_CONTEXT_STACK_DEPTH];
 	size_t top;
diff --git a/lib/crau/crau.h b/lib/crau/crau.h
index 7f75112..96f1df0 100644
--- a/lib/crau/crau.h
+++ b/lib/crau/crau.h
@@ -264,6 +264,11 @@ void crau_data(struct crau_context_stack_st *stack, ...)
 #   endif
 #  endif /* CRAU_MAYBE_UNUSED */
 
+/* fallback if none of the above works */
+#ifndef CRAU_MAYBE_UNUSED
+#define CRAU_MAYBE_UNUSED __attribute__((unused))
+#endif
+
 void crau_push_context(struct crau_context_stack_st *stack CRAU_MAYBE_UNUSED,
 		       long context CRAU_MAYBE_UNUSED)
 {
