diff --git a/Makefile b/Makefile
index 3a63885..65f4b9c 100644
--- a/Makefile
+++ b/Makefile
@@ -249,6 +249,10 @@ else ifneq (,$(findstring odroid64,$(platform)))
       GLES = 0
       GLES3= 1
       GL_LIB := -lGLESv3
+   else ifneq (,$(findstring ODROIDGOA,$(BOARD)))
+      # ODROIDGOA(aarch64)
+      CPUFLAGS += -mcpu=cortex-a35 -mtune=cortex-a35
+      GLES3 = 1
    endif
 
    COREFLAGS += -DOS_LINUX
