diff --git a/Source/Core/Core/Debugger/PPCDebugInterface.cpp b/Source/Core/Core/Debugger/PPCDebugInterface.cpp
index f44c20d..6beda8d 100644
--- a/Source/Core/Core/Debugger/PPCDebugInterface.cpp
+++ b/Source/Core/Core/Debugger/PPCDebugInterface.cpp
@@ -7,7 +7,7 @@
 #include <array>
 #include <cstddef>
 #include <string>
-
+#include <algorithm>
 #include <fmt/format.h>
 
 #include "Common/Align.h"
