diff -uNr a/yabause/src/sh2core.h b/yabause/src/sh2core.h
--- a/yabause/src/sh2core.h	2025-04-01 07:15:45.000000000 +0900
+++ b/yabause/src/sh2core.h	2025-04-01 08:01:06.129079514 +0900
@@ -562,7 +562,7 @@
 void FASTCALL SH2OnFrame(SH2_struct *context);
 
 void SH2RemoveInterrupt(SH2_struct *context, u8 vector, u8 level);
-
+void SH2HandleInterrupts(SH2_struct *context);
 
 void FASTCALL SH2OnFrame(SH2_struct *context);
 
