diff -uNr a/yabause/src/scsp.c b/yabause/src/scsp.c
--- a/yabause/src/scsp.c	2025-04-01 07:15:45.000000000 +0900
+++ b/yabause/src/scsp.c	2025-04-01 08:14:34.978039477 +0900
@@ -5689,14 +5689,14 @@
         }
         pthread_mutex_unlock(&sync_mutex);
 #elif defined(ARCH_IS_LINUX)    
-        time(&tm);    
+        time((time_t *)&tm);    
         long n = tm.tv_nsec;
         tm.tv_nsec += sleeptime;
         if( n > tm.tv_nsec){
           tm.tv_sec += 1;
         }
         pthread_mutex_lock(&sync_mutex);
-        int rtn = pthread_cond_timedwait(&sync_cnd,&sync_mutex,ctime(&tm));
+        int rtn = pthread_cond_timedwait(&sync_cnd,&sync_mutex,(const struct timespec * restrict)ctime((time_t *)&tm));
         if(rtn == 0){
           for (i = 0; i < samplecnt; i += step) {
             MM68KExec(step);
