diff --git a/pcsx2/CDVD/CDVDdiscReader.cpp b/pcsx2/CDVD/CDVDdiscReader.cpp
index 9eb7830..f2aaee2 100644
--- a/pcsx2/CDVD/CDVDdiscReader.cpp
+++ b/pcsx2/CDVD/CDVDdiscReader.cpp
@@ -20,6 +20,7 @@
 #include "AppConfig.h"
 
 #include <condition_variable>
+#include <system_error>
 
 void (*newDiscCB)(void);
 
diff --git a/pcsx2/CDVD/CDVDdiscReader.h b/pcsx2/CDVD/CDVDdiscReader.h
index 6d1078e..dee2f8d 100644
--- a/pcsx2/CDVD/CDVDdiscReader.h
+++ b/pcsx2/CDVD/CDVDdiscReader.h
@@ -26,6 +26,7 @@
 #include <array>
 #include <memory> /* for unique_ptr */
 #include <vector>
+#include <string>
 
 #include "Pcsx2Defs.h"
 
diff --git a/pcsx2/CDVD/CDVDdiscThread.cpp b/pcsx2/CDVD/CDVDdiscThread.cpp
index 3619b82..19f030e 100644
--- a/pcsx2/CDVD/CDVDdiscThread.cpp
+++ b/pcsx2/CDVD/CDVDdiscThread.cpp
@@ -22,6 +22,7 @@
 #include <queue>
 #include <cstring> /* memcpy */
 #include <thread>
+#include <system_error>
 
 const u32 sectors_per_read = 16;
 
diff --git a/pcsx2/MemoryPatchDatabase.h b/pcsx2/MemoryPatchDatabase.h
index 6d0d17c..cf6d60e 100644
--- a/pcsx2/MemoryPatchDatabase.h
+++ b/pcsx2/MemoryPatchDatabase.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <unordered_map>
 #include <string>
 #include <vector>
