diff -uNr a/board/sunxi/board.c b/board/sunxi/board.c
--- a/board/sunxi/board.c	2025-02-12 19:06:45.428754300 +0900
+++ b/board/sunxi/board.c	2025-02-12 19:12:33.888700128 +0900
@@ -225,6 +225,15 @@
 	if (ret)
 		return ret;
 
+#if CONFIG_MACH_SUN50I_H616
+	/*
+	 * The bit[16] of register reg[0x03000000] must be zero for the THS
+	 * driver to work properly in the kernel. The BSP u-boot is putting
+	 * the whole register to zero so we are doing the same.
+	 */
+	writel(0x0, SUNXI_SRAMC_BASE);
+#endif
+
 	eth_init_board();
 
 	return 0;
