diff -uNr a/include/libfdt.h b/include/libfdt.h
--- a/include/libfdt.h	2023-01-09 16:10:25.000000000 +0900
+++ b/include/libfdt.h	2026-09-08 08:13:35.096693123 +0900
@@ -1,5 +1,5 @@
-#ifndef _LIBFDT_H
-#define _LIBFDT_H
+#ifndef LIBFDT_H
+#define LIBFDT_H
 /*
  * libfdt - Flat Device Tree manipulation
  * Copyright (C) 2006 David Gibson, IBM Corporation.
diff -uNr a/include/libfdt_env.h b/include/libfdt_env.h
--- a/include/libfdt_env.h	2026-09-08 08:04:15.517489500 +0900
+++ b/include/libfdt_env.h	2026-09-08 08:08:24.418113656 +0900
@@ -6,8 +6,8 @@
  * SPDX-License-Identifier:	LGPL-2.1+
  */
 
-#ifndef _LIBFDT_ENV_H
-#define _LIBFDT_ENV_H
+#ifndef LIBFDT_ENV_H
+#define LIBFDT_ENV_H
 
 #include "compiler.h"
 #include "linux/types.h"
