diff -uNr a/include/configs/odroidgou.h b/include/configs/odroidgou.h
--- a/include/configs/odroidgou.h	2026-09-08 08:36:32.090637400 +0900
+++ b/include/configs/odroidgou.h	2026-09-08 12:40:01.207813999 +0900
@@ -271,6 +271,7 @@
 #define CONFIG_PTABLE_OFFSET			(CONFIG_ENV_OFFSET + CONFIG_ENV_SIZE)
 
 #define CONFIG_FIT				1
+#define CONFIG_FIT_SIGNATURE			1
 #define CONFIG_OF_LIBFDT			1
 #define CONFIG_OF_LIBFDT_OVERLAY		1
 #define CONFIG_ANDROID_BOOT_IMAGE		1
