diff --git a/src/mnl.c b/src/mnl.c
index 0a44518..0c0f079 100644
--- a/src/mnl.c
+++ b/src/mnl.c
@@ -1570,7 +1570,7 @@ static void obj_tunnel_add_opts(struct nftnl_obj *nlo, struct tunnel *tunnel)
 		nftnl_tunnel_opts_add(opts, opt);
 		nftnl_obj_set_data(nlo, NFTNL_OBJ_TUNNEL_OPTS, &opts, sizeof(struct nftnl_tunnel_opts *));
 		break;
-	case TUNNEL_GENEVE:
+	case TUNNEL_GENEVE: {
 		struct tunnel_geneve *geneve;
 
 		opts = nftnl_tunnel_opts_alloc(NFTNL_TUNNEL_TYPE_GENEVE);
@@ -1594,7 +1594,7 @@ static void obj_tunnel_add_opts(struct nftnl_obj *nlo, struct tunnel *tunnel)
 			nftnl_tunnel_opts_add(opts, opt);
 		}
 		nftnl_obj_set_data(nlo, NFTNL_OBJ_TUNNEL_OPTS, &opts, sizeof(struct nftnl_tunnel_opts *));
-		break;
+		break; }
 	case TUNNEL_UNSPEC:
 		break;
 	}
diff --git a/src/netlink.c b/src/netlink.c
index 26cf07c..09c9d13 100644
--- a/src/netlink.c
+++ b/src/netlink.c
@@ -1881,7 +1881,7 @@ static int tunnel_parse_opt_cb(struct nftnl_tunnel_opt *opt, void *data) {
 			obj->tunnel.vxlan.gbp = nftnl_tunnel_opt_get_u32(opt, NFTNL_TUNNEL_VXLAN_GBP);
 		}
 		break;
-	case NFTNL_TUNNEL_TYPE_GENEVE:
+	case NFTNL_TUNNEL_TYPE_GENEVE: {
 		struct tunnel_geneve *geneve;
 		const void *data;
 
@@ -1909,7 +1909,7 @@ static int tunnel_parse_opt_cb(struct nftnl_tunnel_opt *opt, void *data) {
 		}
 
 		list_add_tail(&geneve->list, &obj->tunnel.geneve_opts);
-		break;
+		break; }
 	default:
 		break;
 	}
diff --git a/src/rule.c b/src/rule.c
index 8f8b77f..c603dcf 100644
--- a/src/rule.c
+++ b/src/rule.c
@@ -2118,7 +2118,7 @@ static void obj_print_data(const struct obj *obj,
 			nft_print(octx, "%s%s%s}",
 				  opts->nl, opts->tab, opts->tab);
 			break;
-		case TUNNEL_GENEVE:
+		case TUNNEL_GENEVE: {
 			struct tunnel_geneve *geneve;
 
 			nft_print(octx, "%s%s%sgeneve {", opts->nl, opts->tab, opts->tab);
@@ -2138,7 +2138,7 @@ static void obj_print_data(const struct obj *obj,
 
 			}
 			nft_print(octx, "%s%s%s}", opts->nl, opts->tab, opts->tab);
-			break;
+			break; }
 		default:
 			break;
 		}
