diff -uNr a/desmume/src/frontend/libretro/Makefile.libretro b/desmume/src/frontend/libretro/Makefile.libretro
--- a/desmume/src/frontend/libretro/Makefile.libretro	2026-05-06 08:24:38.959570600 +0900
+++ b/desmume/src/frontend/libretro/Makefile.libretro	2026-05-06 08:51:09.644494976 +0900
@@ -81,10 +81,10 @@
          DESMUME_JIT_ARM = 1
          DESMUME_JIT = 0
       else
-         LIBS := -lpthread -lGL -lpcap
-         CXXFLAGS += -DHAVE_OPENGL -std=gnu++11
-         DESMUME_OPENGL = 1
-         DESMUME_OPENGL_CORE = 1
+         LIBS := -lpthread -lpcap
+         CXXFLAGS += -std=gnu++11
+         # DESMUME_OPENGL = 1
+         # DESMUME_OPENGL_CORE = 1
       endif
    endif
 
