diff -ruN a/Makefile.common b/Makefile.common
--- a/Makefile.common
+++ b/Makefile.common
@@ -2976,6 +2976,10 @@
          gfx/drivers/oga_gfx.o
 endif
 
+ifeq ($(HAVE_LAKKA_BACKLIGHT), 1)
+  DEFINES += -DHAVE_LAKKA_BACKLIGHT
+endif
+
 ifeq ($(HAVE_GAME_AI),1)
    DEFINES += -DHAVE_GAME_AI
    OBJ += ai/game_ai.o
diff -ruN a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -4416,7 +4416,7 @@
 #else
    NULL,                               /* get_lakka_version */
 #endif
-#if defined(HAVE_LAKKA_SWITCH) || (defined(HAVE_LAKKA) && defined(HAVE_ODROIDGO2))
+#if defined(HAVE_LAKKA_SWITCH) || (defined(HAVE_LAKKA) && (defined(HAVE_ODROIDGO2) || defined(HAVE_LAKKA_BACKLIGHT)))
    frontend_unix_set_screen_brightness,/* set_screen_brightness */
 #else
    NULL,                         /* set_screen_brightness */
diff -ruN a/qb/config.params.sh b/qb/config.params.sh
--- a/qb/config.params.sh
+++ b/qb/config.params.sh
@@ -216,6 +216,7 @@
 HAVE_STEAM=no              # Enable Steam build
 HAVE_MIST=no               # Enable Steam build w/ mist
 HAVE_ODROIDGO2=no          # ODROID-GO Advance rotation support (requires librga)
+HAVE_LAKKA_BACKLIGHT=no    # Lakka handhelds: screen brightness via /sys/class/backlight/backlight
 HAVE_LIBSHAKE=no           # libShake haptic feedback support
 HAVE_CHECK=no              # check support for unit tests
 HAVE_WIFI=no               # wifi driver support
