diff -uNr a/configuration.c b/configuration.c
--- a/configuration.c	2026-08-26 01:48:28.211919948 +0000
+++ b/configuration.c	2026-08-26 01:48:39.479128983 +0000
@@ -73,7 +73,6 @@
 
 #ifdef HAVE_LAKKA_SWITCH
 #include "lakka-switch.h"
-   }
 #endif
 
 #if defined(HAVE_LIBNX)
@@ -7077,6 +7076,7 @@
       fprintf(f, "0\n");
       fclose(f);
    }
+   }
 #endif
 
    frontend_driver_set_sustained_performance_mode(settings->bools.sustained_performance_mode);
