diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4cb1106..cf41e41 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -240,8 +240,14 @@ endif()
 if(LIBRETRO)
 	add_compile_definitions(__LIBRETRO__)
 	add_compile_definitions(GLEW_NO_GLU)
+	if(ARM64)
+		add_compile_options(-mno-outline-atomics)
+	endif()
 	if(NOT MSVC)
 		add_compile_options(-fPIC)
+		if(NOT ANDROID)
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+		endif()
 	endif()
 endif()
 
