diff -r -u a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts
--- a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts	2024-07-01 16:54:38.266449300 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351m.dts	2024-07-01 21:53:38.596754000 +0900
@@ -6,66 +6,53 @@
  */
 
 /dts-v1/;
-#include "rk3326-anbernic-rg351m.dtsi"
+#include "rk3326-odroid-go.dtsi"
 
 / {
 	model = "Anbernic RG351M";
 	compatible = "anbernic,rg351m", "rockchip,rk3326";
 
-	battery: battery {
-			compatible = "simple-battery";
-			charge-full-design-microamp-hours = <3500000>;
-			charge-term-current-microamp = <350000>;
-			constant-charge-current-max-microamp = <2000000>;
-			constant-charge-voltage-max-microvolt = <4200000>;
-			factory-internal-resistance-micro-ohms = <180000>;
-			voltage-max-design-microvolt = <4100000>;
-			voltage-min-design-microvolt = <3500000>;
-
-			ocv-capacity-celsius = <20>;
-			ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
-									<3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
-									<3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
-									<3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
-									<3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
-									<3574170 0>;
-	};
-
-	rk817-sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,name = "rk817_int";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,hp-det-gpio = <&gpio2 RK_PC6 GPIO_ACTIVE_HIGH>;
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,widgets =
-			"Microphone", "Mic Jack",
-			"Headphone", "Headphones",
-			"Speaker", "Speaker";
-		simple-audio-card,routing =
-			"MICL", "Mic Jack",
-			"Headphones", "HPOL",
-			"Headphones", "HPOR",
-			"Speaker", "SPKO";
-
-		simple-audio-card,codec {
-			sound-dai = <&rk817>;
-		};
-
-		simple-audio-card,cpu {
-			sound-dai = <&i2s1_2ch>;
-		};
+        battery: battery {
+               compatible = "simple-battery";
+               charge-full-design-microamp-hours = <3500000>;
+               charge-term-current-microamp = <350000>;
+               constant-charge-current-max-microamp = <2000000>;
+               constant-charge-voltage-max-microvolt = <4200000>;
+               factory-internal-resistance-micro-ohms = <180000>;
+               voltage-max-design-microvolt = <4100000>;
+               voltage-min-design-microvolt = <3500000>;
+
+               ocv-capacity-celsius = <20>;
+               ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
+                                       <3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
+                                       <3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
+                                       <3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
+                                       <3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
+                                       <3574170 0>;
+       };
+
+	vibrator {
+		compatible = "pwm-vibrator";
+		pwms = <&pwm0 0 1000000 0>;
+		pwm-names = "enable";
 	};
 };
 
+/delete-node/ &vcc_host; /* conflicts with pwm vibration motor */
+
 &internal_display {
 	compatible = "elida,kd35t133";
 	iovcc-supply = <&vcc_lcd>;
-	rotation = <270>;
 	vdd-supply = <&vcc_lcd>;
+	rotation = <270>;
+};
+
+&pwm0 {
+	status = "okay";
 };
 
 &rk817_charger {
-	/* This device does not have a current sense resistor */
-	//rockchip,resistor-sense-micro-ohms = <0>;
-	monitored-battery = <&battery>;
-};
\ No newline at end of file
+       /* This device does not have a current sense resistor */
+       //rockchip,resistor-sense-micro-ohms = <0>;
+       monitored-battery = <&battery>;
+};
diff -r -u a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts
--- a/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts	2024-07-01 16:54:38.266449300 +0900
+++ b/arch/arm64/boot/dts/rockchip/rk3326-anbernic-rg351v.dts	2024-07-01 21:53:38.596754000 +0900
@@ -1,81 +1,72 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2019 Hardkernel Co., Ltd
+ * Copyright (c) 2020 Theobroma Systems Design und Consulting GmbH
+ * Copyright (c) 2022 Maya Matuszczyk <maccraft123mc@gmail.com>
+ */
 
 /dts-v1/;
-#include "rk3326-anbernic-rg351m.dtsi"
+#include "rk3326-odroid-go.dtsi"
 
 / {
 	model = "Anbernic RG351V";
 	compatible = "anbernic,rg351v", "rockchip,rk3326";
 
-	gpio-keys-vol {
-		compatible = "gpio-keys";
-		autorepeat;
-		pinctrl-0 = <&btn_pins_vol>;
-		pinctrl-names = "default";
-
-		button-vol-down {
-				gpios = <&gpio2 RK_PA1 GPIO_ACTIVE_LOW>;
-				label = "VOLUMEDOWN";
-				linux,code = <KEY_VOLUMEDOWN>;
-		};
-
-		button-volume-up {
-				gpios = <&gpio2 RK_PA0 GPIO_ACTIVE_LOW>;
-				label = "VOLUMEUP";
-				linux,code = <KEY_VOLUMEUP>;
-		};
+       aliases {
+               mmc0 = &sdio;
+               mmc1 = &sdmmc;
+       };
+
+        gpio_keys: odroidgo3-keys {
+                compatible = "gpio-keys";
+                autorepeat;
+                pinctrl-0 = <&btn_pins_vol>;
+                pinctrl-names = "default";
+
+                button-vol-down {
+                        gpios = <&gpio2 RK_PA1 GPIO_ACTIVE_LOW>;
+                        label = "GPIO BTN-VOLDN";
+                        linux,code = <KEY_VOLUMEDOWN>;
+                };
+
+                button-volume-up {
+                        gpios = <&gpio2 RK_PA0 GPIO_ACTIVE_LOW>;
+                        label = "GPIO BTN-VOLUP";
+                        linux,code = <KEY_VOLUMEUP>;
+                };
+        };
+
+        battery: battery {
+               compatible = "simple-battery";
+               charge-full-design-microamp-hours = <3500000>;
+               charge-term-current-microamp = <350000>;
+               constant-charge-current-max-microamp = <2000000>;
+               constant-charge-voltage-max-microvolt = <4200000>;
+               factory-internal-resistance-micro-ohms = <180000>;
+               voltage-max-design-microvolt = <4100000>;
+               voltage-min-design-microvolt = <3500000>;
+
+               ocv-capacity-celsius = <20>;
+               ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
+                                       <3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
+                                       <3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
+                                       <3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
+                                       <3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
+                                       <3574170 0>;
+       };
+
+	vibrator {
+		compatible = "pwm-vibrator";
+		pwms = <&pwm0 0 1000000 0>;
+		pwm-names = "enable";
 	};
+};
 
-	battery: battery {
-		compatible = "simple-battery";
-		charge-full-design-microamp-hours = <3500000>;
-		charge-term-current-microamp = <350000>;
-		constant-charge-current-max-microamp = <2000000>;
-		constant-charge-voltage-max-microvolt = <4200000>;
-		factory-internal-resistance-micro-ohms = <180000>;
-		voltage-max-design-microvolt = <4100000>;
-		voltage-min-design-microvolt = <3500000>;
-
-		ocv-capacity-celsius = <20>;
-		ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
-								<3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
-								<3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
-								<3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
-								<3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
-								<3574170 0>;
-	};
+/* conflicts with pwm vibration motor */
+/delete-node/ &vcc_host;
 
-	rk817-sound {
-		compatible = "simple-audio-card";
-		pinctrl-names = "default";
-		pinctrl-0 = <&hp_det>;
-
-		simple-audio-card,name = "rk817_int";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,mclk-fs = <256>;
-		
-		simple-audio-card,hp-det-gpio = <&gpio2 RK_PC6 GPIO_ACTIVE_LOW>;
-		simple-audio-card,hp-pin-name = "Headphones";
-		
-		simple-audio-card,widgets =
-			"Microphone", "Mic Jack",
-			"Headphone", "Headphones",
-			"Speaker", "Speaker";
-		simple-audio-card,routing =
-			"MICL", "Mic Jack",
-			"Headphones", "HPOL",
-			"Headphones", "HPOR",
-			"Speaker", "SPKO";
-
-		simple-audio-card,codec {
-			sound-dai = <&rk817>;
-		};
-
-		simple-audio-card,cpu {
-			sound-dai = <&i2s1_2ch>;
-		};
-	};
-};
+/* Device only has 1 LED compared to Odroid Go Advance */
+/delete-node/ &gpio_led;
 
 &internal_display {
 	compatible = "anbernic,rg351v-panel", "newvision,nv3051d";
@@ -87,26 +78,47 @@
 	vccio1-supply = <&vccio_sd>;
 };
 
-&vcc_sd {
-	regulator-max-microvolt = <3000000>;
-	regulator-min-microvolt = <1800000>;
+/delete-node/ &pwm_led;
+
+&pwm0 {
+	status = "okay";
 };
 
+/delete-node/ &pwm3;
+
 &rk817_charger {
        /* This device does not have a current sense resistor */
        rockchip,resistor-sense-micro-ohms = <0>;
        monitored-battery = <&battery>;
 };
 
+&sdio {
+	cap-sd-highspeed;
+	card-detect-delay = <200>;
+	cd-gpios = <&gpio3 RK_PB4 GPIO_ACTIVE_LOW>; /*[> CD GPIO <]*/
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	vmmc-supply = <&vcc_sd>;
+	vqmmc-supply = <&vccio_sd>;
+	status = "okay";
+};
+
+&vcc_sd {
+	regulator-max-microvolt = <3000000>;
+	regulator-min-microvolt = <1800000>;
+};
+
 &vccio_sd {
 	regulator-max-microvolt = <1800000>;
 };
 
 &pinctrl {
-	btns {
-		btn_pins_vol: btn-pins-vol {
-			rockchip,pins = <2 RK_PA0 RK_FUNC_GPIO &pcfg_pull_up>,
-							<2 RK_PA1 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-};
\ No newline at end of file
+        btns {
+                btn_pins_vol: btn-pins-vol {
+                        rockchip,pins = <2 RK_PA0 RK_FUNC_GPIO &pcfg_pull_up>,
+                                        <2 RK_PA1 RK_FUNC_GPIO &pcfg_pull_up>;
+                };
+        };
+};
