--- a/src/amd/vulkan/meson.build
+++ b/src/amd/vulkan/meson.build
@@ -296,7 +296,7 @@
   c_args : [no_override_init_args, radv_flags, c_msvc_compat_args],
   cpp_args : [radv_flags, cpp_msvc_compat_args],
   link_args : [
-    ld_args_build_id, ld_args_bsymbolic, ld_args_gc_sections, vulkan_icd_link_args,
+    ld_args_build_id, ld_args_bsymbolic, ld_args_gc_sections, vulkan_icd_link_args, '-lgcc',
   ],
   link_depends : vulkan_icd_link_depends,
   gnu_symbol_visibility : 'hidden',
