diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index 8555a2f8..5b23cf65 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -311,15 +311,10 @@ struct retro_core_option_v2_definition option_defs_us[] = {
 #if defined(IOS)
         "cached_interpreter"
 #else
-#ifdef DYNAREC
-#if defined(HAVE_DYNAREC_HACKTARUX) && defined(NEW_DYNAREC)
-        "dynamic_recompiler_ari64"
-#else
-        "dynamic_recompiler"
-#endif
-#else
+        /* RPi5 fails to launch parallel_n64 core.
+         * Therefore, it uses "cached_interpreter" for "parallel-n64-cpucore" as workaround.
+         * https://github.com/libretro/Lakka-LibreELEC/issues/1982#issuecomment-2154720599 */
         "cached_interpreter"
-#endif
 #endif
     },
     {
